includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -1881,4 +1903,31 @@ public function get_button_locations() {
 
 		return $this->stripe_settings['payment_request_button_locations'];
 	}
+
+
+	/**
+	 * Populate $_POST with all form input data to allow 3rd party code to validate.
+	 *
+	 * @since   7.0.0
+	 */
+	private function populate_post() {
+		if ( isset( $_POST['custom_data'] ) ) {
+
+			parse_str( $_POST['custom_data'], $custom_data );
+
+			if ( $custom_data ) {
+
+				foreach ( $custom_data as $input_name => $input_value ) {
+
+					// Write to $_POST only if key does not exist
+					if ( ! isset( $_POST[ $input_name ] ) ) {
+						$_REQUEST[ $input_name ] = $input_value;
+						$_POST[ $input_name ]    = $input_value;
```

Comment:
I think it's dangerous to do this. Either way we decide on a fixed name and put all the data you need there or find another approach.

Reply:
You might be right. Just borrowing from how One Page Checkout has traditionally handled relaying 3rd party fields.